src/modules/adaptors/Convex/ConvexAdaptor.sol

```diff
@@ -0,0 +1,268 @@
+     */
+    function withdrawableFrom(
+        bytes memory adaptorData,
+        bytes memory configurationData
+    ) public view override returns (uint256) {
+        bool isLiquid = abi.decode(configurationData, (bool));
+        if (isLiquid) {
+            (, address rewardPool) = abi.decode(adaptorData, (uint256, address));
+            IBaseRewardPool baseRewardPool = IBaseRewardPool(rewardPool);
+            ERC20 stakingToken = ERC20(baseRewardPool.stakingToken());
+            return (stakingToken.balanceOf(msg.sender) + baseRewardPool.balanceOf(msg.sender));
+        } else return 0;
+    }
+
+    /**
+     * @notice Calculates the Cellar's balance of the positions creditAsset, a specific underlying LPT.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @return total balance of LPT for Cellar, including liquid and staked
+     * TODO: This assumes that no rewards are given back as accrual of more curveLPT. I believe that to be the case because BaseRewardPool has its own rewardsToken, and extraRewards has specific reward contracts specific to respective convex markets.
+     */
+    function balanceOf(bytes memory adaptorData) public view override returns (uint256) {
+        (, address rewardPool) = abi.decode(adaptorData, (uint256, address));
+        IBaseRewardPool baseRewardPool = IBaseRewardPool(rewardPool);
+        ERC20 stakingToken = ERC20(baseRewardPool.stakingToken());
+        return (stakingToken.balanceOf(msg.sender) + baseRewardPool.balanceOf(msg.sender));
+    }
+
+    /**
+     * @notice Returns the positions underlying assets.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @return Underlying LPT for Cellar's respective Convex market position
+     */
+    function assetOf(bytes memory adaptorData) public view override returns (ERC20) {
+        (uint256 pid, ) = abi.decode(adaptorData, (uint256, address));
+        (address lpToken, , , , , ) = booster.poolInfo(pid);
+        ERC20 lpt = ERC20(lpToken);
+        return lpt;
+        // TODO: decide to use above or alternative way to get lpt below
+        // IBaseRewardPool rewardsPool = IBaseRewardPool(rewardsPool);
+        // return ERC(rewardsPool.stakingToken());
+    }
+
+    /**
+     * @notice When positions are added to the Registry, this function can be used in order to figure out
+     *         what assets this adaptor needs to price, and confirm pricing is properly setup.
+     * @param adaptorData see adaptorData info at top of this smart contract
+     * @return assets Underlying assets for Cellar's respective Convex market position
+     * @dev all breakdowns of LPT pricing and its underlying assets are done through the PriceRouter extension (in accordance to PriceRouterv2 architecture)
+     */
+    function assetsUsed(bytes memory adaptorData) public view override returns (ERC20[] memory assets) {
```

Comment:
Do you need to override this? Or is this already what BaseAdaptor uses for its implementation?

Reply:
You're right, I've removed it since it simply inherits the `BaseAdaptor`